cms/djangoapps/maintenance/tests.py

```diff
@@ -0,0 +1,242 @@
+"""
+Tests for the maintenance app views.
+"""
+import ddt
+
+from django.core.urlresolvers import reverse
+
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+
+from contentstore.management.commands.utils import get_course_versions
+from student.tests.factories import AdminFactory, UserFactory
+
+from .views import COURSE_KEY_ERROR_MESSAGES, MAINTENANCE_VIEWS
+
+
+# This list contains URLs of all maintenance app views.
+MAINTENANCE_URLS = [view['url'] for view in MAINTENANCE_VIEWS.values()]
+
+
+class TestMaintenanceIndex(ModuleStoreTestCase):
+    """
+    Tests for maintenance index view.
+    """
+
+    def setUp(self):
+        """Make the user global staff."""
+        super(TestMaintenanceIndex, self).setUp()
+        self.user = AdminFactory()
+        login_success = self.client.login(username=self.user.username, password='test')
+        self.assertTrue(login_success)
+        self.view_url = reverse('maintenance:maintenance')
+
+    def test_maintenance_index(self):
+        """Test that maintenance index view lists all the maintenance app views."""
+        response = self.client.get(self.view_url)
+        self.assertContains(response, 'Maintenance', status_code=200)
+
+        # Check that all the expected links appear on the index page.
+        for url in MAINTENANCE_URLS:
+            self.assertContains(response, url, status_code=200)
+
+
+@ddt.ddt
+class MaintenanceViewTestCase(ModuleStoreTestCase):
+    """
+    Base class for maintenance view tests.
+    """
+    view_url = ''
+
+    def setUp(self):
+        """Create a user and log in."""
+        super(MaintenanceViewTestCase, self).setUp()
+        self.user = AdminFactory()
+        login_success = self.client.login(username=self.user.username, password='test')
+        self.assertTrue(login_success)
+        self.course = CourseFactory.create(default_store=ModuleStoreEnum.Type.split)
```

Comment:
Don't make the course here... course creation can slow down tests, so only use it when you need it!